pkg/template/controller/templateinstance_controller.go

```diff
@@ -437,13 +437,17 @@ func (c *TemplateInstanceController) instantiate(templateInstance *templatev1.Te
 			continue
 		}
 
-		namespace := templateInstance.Namespace
-		if len(currObj.GetNamespace()) > 0 {
-			namespace = currObj.GetNamespace()
-		}
-		if namespace == "" {
-			allErrors = append(allErrors, errors.New("namespace was empty"))
-			continue
+		namespace := ""
+		clusterScoped := restMapping.Scope.Name() == meta.RESTScopeNameRoot
+		if !clusterScoped {
+			namespace = templateInstance.Namespace
+			if len(currObj.GetNamespace()) > 0 {
+				namespace = currObj.GetNamespace()
+			}
+			if namespace == "" {
+				allErrors = append(allErrors, errors.New("namespace was empty"))
+				continue
+			}
```

Comment:
@gabemontero Can we DRY this out into a separate function?

Reply:
Yeah it was a little klunky to me because of the continue above 

I'll add it back in and we can confer on how it looks

Reply:
ok a bunch of parameters but the logic is in a separate function @adambkaplan  ... PTAL